internal/handler/buyer.go

```diff
@@ -171,3 +171,47 @@ func (bh *BuyerHandler) HandlerCreateBuyer(w http.ResponseWriter, r *http.Reques
 	}
 
 }
+
+func (bh *BuyerHandler) HandlerUpdateBuyer(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+
+	if err != nil {
+		http.Error(w, "Invalid ID", http.StatusBadRequest)
+		return
+	}
+
+	var reqBody RequestBuyerJson
+
+	err = json.NewDecoder(r.Body).Decode(&reqBody)
+
+	if err != nil {
+		http.Error(w, "erro de desserialização dos dados", http.StatusBadRequest)
+		return
+	}
+
+	buyer, err := bh.svc.UpdateBuyer(id, model.Buyer{
+		CardNumberId: reqBody.CardNumberId,
+		FirstName:    reqBody.FirstName,
+		LastName:     reqBody.LastName,
+	})
+
+	if err != nil && errors.Is(err.(*custom_error.CustomError).Err, custom_error.NotFound) {
+		http.Error(w, "", http.StatusNotFound)
+		return
+	}
```

Comment:
se der algum erro que não seja o notfound, ele nao vai cair nessa validação, fiz isso aqui e deu bom:

```suggestion
if err != nil {
		if errors.Is(err.(custom_error.CustomError).Err, custom_error.NotFound) {
			response.JSON(w, http.StatusNotFound, ResponseBodyError{Status: "error", Message: err.Error()})
			return
		}
		response.JSON(w, http.StatusInternalServerError, ResponseBodyError{Status: "error", Message: err.Error()})
		return
}
```